app/controllers/users_controller.rb

```diff
@@ -58,6 +58,10 @@ def update
         end
       end
 
+      if role_change = @user.previous_changes[:role]
```

Comment:
Is there a good place to test this code as well? There's probably enough logic there to warrant an integration or controller test.